File: SecondYear/ApoorvaSri123_ApoorvaSrivastava_2226it1116_2/Week_3/Move_all_negative_element_at_end.c
#include <stdio.h>

void rearrangeArray(int arr[], int n) {
    int negativeIndex = n - 1;

    for (int i = n - 1; i >= 0; i--) {
        if (arr[i] < 0) {
            int temp = arr[i];
            arr[i] = arr[negativeIndex];
            arr[negativeIndex] = temp;
            negativeIndex--;
        }
    }
}

int main() {
    int n;
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    rearrangeArray(arr, n);

    printf("Modified array with negative elements at the end: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: SecondYear/Divyansh670_DivyanshSrivastav_2226it1029_2/Week_3/Move_all_negative_element_at_end.java
import java.util.Scanner;

public class Move_all_negative_element_at_end {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        scanner.close();

        rearrangeArray(arr, n);

        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    static void rearrangeArray(int[] arr, int n) {
        int negativeIndex = n - 1;

        for (int i = n - 1; i >= 0; i--) {
            if (arr[i] < 0) {
                int temp = arr[i];
                arr[i] = arr[negativeIndex];
                arr[negativeIndex] = temp;
                negativeIndex--;
            }
        }
    }
}
